lib/fileOperations.js

```diff
@@ -196,11 +198,55 @@ function writeFile(path, data, options, callback) {
   }
 }
 
+function mkdirp(dirpath, mode, callback) {
+  if (typeof mode === 'function') {
+    callback = mode;
+    mode = undefined;
+  }
+
+  var m = mode || DEFAULT_DIR_MODE;
+  var cb = callback || function() {};
+  dirpath = path.resolve(dirpath);
+
+  fs.mkdir(dirpath, m, function(er) {
```

Comment:
The mkdir back down the path is on line 221.  It's actually another mkdirp call.

One clarifying question on mode:  
When mode is provided, all newly created directories will be created with that mode.  And currently, if the final directory on the path already exists, then chmod will be called to change the mode.

Do you want ALL existing directories in the provided dirpath parameter to change permission?